api/qan/collector.proto

```diff
@@ -0,0 +1,203 @@
+syntax = "proto3";
+
+package qan;
+
+option go_package = "qan";
+
+// Agent service contains RPCs to implement communication mechanism with qan-agent.
+// Namely query metrics, query explain etc.
+service Agent {
+  // DataInterchange defines duplex communication between Query analytics API and Agent.
+  rpc DataInterchange(stream AgentMessage) returns (stream ApiMessage);
+}
+
+// AgentMessage contains several Query Classes for one or more periods of time.
+message AgentMessage {
+  repeated QueryClass query_class = 1;
+}
+
+// QueryClass is aggregated message created by qan-agent.
+// Contains information about one query selected in defined way from query class in specific period of time.
+message QueryClass {
```

Comment:
for now we are not focused on Queries only. Let's have this  as  "Data bucket " or something like this. but this is not about  Olny Queries now

Reply:
Agree. Renamed.